src/log.h

```diff
@@ -98,13 +101,38 @@ class LogStream
   std::ostringstream buf_;
 };
 
+class LogStreamFatal : public LogStream
+{
+public:
+  LogStreamFatal(const std::string& file,
+                 int line,
+                 __attribute__((unused)) LogType,
+                 std::ostream& out = std::cerr)
+      : LogStream(file, line, LogType::FATAL, out){};
+  LogStreamFatal(const std::string& file,
+                 int line,
+                 __attribute__((unused)) LogType,
+                 const location& loc,
+                 std::ostream& out = std::cerr)
+      : LogStream(file, line, LogType::FATAL, loc, out){};
+  [[noreturn]] ~LogStreamFatal();
+};
+
 // Usage examples:
 // 1. LOG(WARNING) << "this is a " << "warning!"; (this goes to std::cerr)
 // 2. LOG(DEBUG, std::cout) << "this is a " << " message.";
 // 3. LOG(ERROR, call.loc, std::cerr) << "this is a semantic error";
 // Note: LogType::DEBUG will prepend __FILE__ and __LINE__ to the debug message
 
-#define LOG(...) LogStream(__FILE__, __LINE__, LogType::__VA_ARGS__)
+// clang-format off
+#define LOGSTREAM_DEBUG(...) bpftrace::LogStream(__FILE__, __LINE__, __VA_ARGS__)
+#define LOGSTREAM_INFO(...) bpftrace::LogStream(__FILE__, __LINE__, __VA_ARGS__)
+#define LOGSTREAM_WARNING(...) bpftrace::LogStream(__FILE__, __LINE__, __VA_ARGS__)
+#define LOGSTREAM_ERROR(...) bpftrace::LogStream(__FILE__, __LINE__, __VA_ARGS__)
```

Comment:
Can you combine the definitions into a common macro? Not sure if possible, but would be nice if it is.

ie.
```
#define LOGSTREAM_COMMON(...) ...
#define LOGSTREAM_DEBUG(...) LOGSTREAM_COMMON(__VA_ARGS__)
```